source/docs/software/vscode-overview/3rd-party-libraries.rst

```diff
@@ -1,92 +1,91 @@
 
-`CTRE Phoenix Toolsuite <https://www.ctr-electronics.com/control-system/hro.html#product_tabs_technical_resources>`__ - Contains CANCoder, Canifier, Pigeon, Talon FX, Talon SRX, and Victor SPX Libraries and Phoenix Tuner program for configuring CTRE CAN devices
+The JSON file allows specification of complex libraries with multiple components (Java, C++, JNI, etc.) and also helps handle some complexities related to simulation. Vendors choosing to provide a remote URL in the JSON also enable users to check for updates from within VS Code.
 
-`Digilent <https://reference.digilentinc.com/dmc-60c/getting-started>`__ - DMC-60C library
+.. note:: The vendor JSON files are actually processed by GradleRIO once they are in your projects ``vendordeps`` folder. If you are using another IDE, you will need to manually create a “vendordeps” folder in your project and copy any desired vendor JSON files from the “wpilib/YYYY” folder (where they should be placed by an offline installer) or download them directly from the vendor and place them into the folder in the project.
 
-`Playing With Fusion Driver <https://www.playingwithfusion.com/docview.php?docid=1205>`__ - Library for all PWF devices including the Venom motor/controller
+The Mechanism - LabVIEW
+^^^^^^^^^^^^^^^^^^^^^^^
 
-`Kauai Labs <https://pdocs.kauailabs.com/navx-mxp/software/roborio-libraries/>`__ - Libraries for NavX-MXP, NavX-Micro, and Sensor Fusion
+For LabVIEW teams, you may notice a few new Third Party items on various palettes (specifically, one in :guilabel:`Actuators`, one in :guilabel:`Actuators` -> :guilabel:`Motor Control` labeled :guilabel:`CAN Motor`, and one in :guilabel:`Sensors`). These correspond to folders in ``C:\Program Files\National Instruments\LabVIEW 2016\vi.lib\Rock Robotics\WPI\Third Party``.
 
-`Rev Robotics Color Sensor V3 <http://www.revrobotics.com/rev-31-1557/>`__ - Library for Rev Robotics Color Sensor included in 2020 Kit of Parts
+To use installed "Third Party" libraries, simply locate the VIs in one of these 3 locations and drag them into your project as you would with any other VI.
 
-`Rev Robotics SPARK MAX <https://www.revrobotics.com/sparkmax-software/>`__ - SPARK MAX Library
+Adding an Offline-Installed Library
```

Comment:
This seems out of place being after the LabVIEW section, but doesn't apply to LabVIEW. I would move Labview up top, then do C++/Java mechanism, then do the vs-code instructions.

In addition, I would move  Command-Line Interface (Online) up here too, it seems out of place. I would be tempted to call it GradleRIO and the others VS Code for clarity.

Reply:
> This seems out of place being after the LabVIEW section, but doesn't apply to LabVIEW. I would move Labview up top, then do C++/Java mechanism, then do the vs-code instructions.

I've made a "Managing VS Code Libraries" section that wraps in the VS Code instructions. I disagree with moving LabVIEW up above Java/C++ as this can cause users to just not scroll down and ignore the fact that there is Java/C++ instructions still there. 

> In addition, I would move Command-Line Interface (Online) up here too, it seems out of place. 

I'm not sure what you mean, it seems fine to me.

>  I would be tempted to call it GradleRIO and the others VS Code for clarity.

I disagree with calling it GradleRIO as we don't call any other command line instructions (like ``./gradlew`` as gradlerio commands and just the command line.). And I think wrapping the VS Code instructions improves clarity.